src/components/Sidebars/Generate.js

```diff
@@ -124,22 +62,20 @@ const Generate = ({ cdpId, reset }) => {
       <InfoContainer>
         <Info
           title={lang.action_sidebar.maximum_available_to_generate}
-          body={`${safeToFixed(daiAvailable, 7)} DAI`}
+          body={cdp.daiAvailable().toString()}
         />
         <Info
           title={lang.action_sidebar.new_liquidation_price}
-          body={formatLiquidationPrice(liquidationPrice, cdp.currency.symbol)}
+          body={cdp.liquidationPrice({ dart: amountToGenerate }).toString()}
         />
         <Info
           title={lang.action_sidebar.new_collateralization_ratio}
           body={
-            <RatioDisplay
-              type={RatioDisplayTypes.TEXT}
-              ratio={collateralizationRatio}
-              ilkLiqRatio={liquidationRatio}
-              text={formatCollateralizationRatio(collateralizationRatio)}
-              show={amount !== '' && amount > 0 && !undercollateralized}
-            />
+            <Text color={cdpUnderCollateralized ? 'orange.600' : null}>
+              {cdp
+                .collateralizationRatio({ dart: amountToGenerate })
+                .toString()}
+            </Text>
```

Comment:
Just with the colours, I've added a [function](https://github.com/makerdao/mcd-cdp-portal/blob/8a13ff8a573450eb3a5fdf272e343eee90398a7c/src/styles/theme.js#L81) in themes.js that centralizes a colour scheme for several components that uses safety levels within the app. I feel like we should derive colours based on the underlying values where we can?

One thought I had looking at this PR is we could remove `ratio` and `ilkLiqRatio` as args from `RatioDisplay` and pass in the cdp object directly?

Reply:
>  I feel like we should derive colours based on the underlying values where we can?
👍👍

> One thought I had looking at this PR is we could remove ratio and ilkLiqRatio as args from RatioDisplay and pass in the cdp object directly?
ah yep. another option is to just pass the `cdp id` in and have `RatioDisplay` rely on a `useCdp` hook under the hood.

Reply:
Perfect, better again!